components/shared/Search.tsx

```diff
@@ -1,33 +1,158 @@
+
+  useEffect(() => {
+    const getPosts = async () => {
+      const posts = await findPosts(searchTerm);
+      if (posts) setPosts(posts as Post[]);
+    };
+    getPosts();
+  }, [searchTerm]);
 
-export default function Search() {
-  function handleSearch(term: string) {
-    console.log(term);
-  }
+  useEffect(() => {
+    const timeout = setTimeout(() => {
+      const newParams = urlManager(searchParams.toString(), {
+        page: "1",
+        term: searchTerm,
+      });
+      router.push(`/?${newParams}`);
+    }, 500);
+    return () => clearTimeout(timeout);
+  }, [searchTerm]);
 
+  // Toggle the menu when ⌘K is pressed
+  useEffect(() => {
+    const down = (e: {
+      key: string;
+      metaKey: any;
+      ctrlKey: any;
+      preventDefault: () => void;
+    }) => {
+      if (e.key === "k" && (e.metaKey || e.ctrlKey)) {
+        e.preventDefault();
+        setOpen((open) => !open);
+      }
+    };
+
+    document.addEventListener("keydown", down);
+    return () => document.removeEventListener("keydown", down);
+  }, []);
+
+  const iconMatch = (post: Post) => {
+    switch (post.createType) {
+      case CreateType.COMPONENT:
+        return (
+          <>
+            <ComponentIcon className="text-purple-500" size={18} />
+            <span>{post.title}</span>
+          </>
+        );
+      case CreateType.WORKFLOW:
+        return (
+          <>
+            <WorkflowIcon className="text-primary-500" size={18} />
+            <span>{post.title}</span>
+          </>
+        );
+      case CreateType.KNOWLEDGE:
+        return (
+          <>
+            <KnowledgeIcon className="text-green-500" size={18} />
+            <span>{post.title}</span>
+          </>
+        );
+    }
+  };
   return (
-    <div className="bg-black-700 flex items-center gap-2 rounded-md px-2">
-      <Image
-        src={searchIcon}
-        alt="Search Icon"
-        className="pointer-events-none"
-      />
-      <input
-        className="placeholder:paragraph-4-medium placeholder:text-white-500 bg-black-700 w-full rounded-md border-none text-sm focus:outline-none"
-        placeholder="Search..."
-        onChange={(e) => {
-          handleSearch(e.target.value);
-        }}
-      />
-      <Image
-        src={shortcutIcon}
-        alt="Shortcut Icon"
-        className="pointer-events-none"
-      />
-    </div>
+    <>
+      <div
+        className="bg-black-700 paragraph-4-medium flex cursor-pointer items-center justify-between rounded-md p-4"
+        onClick={() => setOpen((open) => !open)}
+      >
+        <div className="flex gap-x-2">
+          <Image
+            src={searchIcon}
+            alt="Search Icon"
+            className="pointer-events-none"
+          />
+          <p className="text-white-500">Search...</p>
+        </div>
+        <Image
+          src={shortcutIcon}
+          alt="Shortcut Icon"
+          className="pointer-events-none"
+        />
+      </div>
+
+      <Command.Dialog
```

Comment:
attempted to add an outside click handler... but for some reason it wasn't working for me... In a previous branch, I created a custom hook for it -  [https://github.com/DDVVPP/GitNote/blob/SearchLeftNav/lib/utils/useOutsideClickHandler.ts](https://github.com/DDVVPP/GitNote/blob/SearchLeftNav/lib/utils/useOutsideClickHandler.ts)
 
 And tried to implement it like this:
 ```
const dialogRef = useRef<HTMLDivElement>(null);
useOutsideClickHandler(dialogRef, () => setOpen(false));
```
I added the `dialogRef` to `Command.Dialog` but had no luck.  I wonder if there are refs there already that are conflicting with this one?

Reply:
So much for simple - thank you!